kart/spatial_filters.py

```diff
@@ -105,26 +125,41 @@ def matches(self, feature_geometry):
 
     def transform_for_dataset(self, dataset):
         """Transform this spatial filter so that it matches the CRS of the given dataset."""
-        if self.match_all or self.filter_ogr is None or self.crs is None:
-            return self
+        if self.match_all or dataset is None:
+            return SpatialFilter.MATCH_ALL
 
-        ds_path = dataset.path
-        ds_crs_defs = dataset.crs_definitions()
-        if not ds_crs_defs:
-            return self
-        if len(ds_crs_defs) > 1:
-            raise NotYetImplemented(
-                f"Sorry, spatial filtering dataset {ds_path!r} with multiple CRS is not yet supported"
-            )
-        ds_crs_def = list(ds_crs_defs.values())[0]
-        return self.transform_for_crs(ds_crs_def, ds_path)
+        geom_column_name = dataset.geom_column_name
+        if not geom_column_name:
+            return SpatialFilter.MATCH_ALL
+        result = self.with_geom_column_name(geom_column_name)
+
+        if self.crs is not None:
+            ds_path = dataset.path
+            ds_crs_defs = dataset.crs_definitions()
+            if not ds_crs_defs:
+                return self
+            if len(ds_crs_defs) > 1:
+                raise NotYetImplemented(
+                    f"Sorry, spatial filtering dataset {ds_path!r} with multiple CRS is not yet supported"
+                )
+            ds_crs_def = list(ds_crs_defs.values())[0]
+            return result.transform_for_crs(ds_crs_def, ds_path)
+        else:
+            return result
```

Comment:
what does it mean for a spatial filter to have a null CRS? we just assume it's the same as the datasets? Maybe it'd be better to force it to be explicit, or possibly assume it's WGS84

Reply:
I don't think a null CRS is really a good idea: Server-side, we need to get the filter to WGS84 in order to do the clone filtering, and to do that conversion the source needs to be known.

Reply:
Default CRS is now EPSG:4326, if it is not specified in the config.